homeassistant/components/zwave_js/entity.py

```diff
@@ -164,6 +176,22 @@ def _value_changed(self, event_data: dict) -> None:
             value.value,
         )
 
+        if self.info.assumed_state:
+            self.hass.bus.async_fire(
```

Comment:
We should keep track if these devices separately outside of the entity, and fire the event outside any platform.

Reply:
Are you thinking we track this in the device registry entry as well?

Re: outside any platform, that means that we will fire this for all value updates for a given device? This falls outside of the boundaries of discovery then since that's done on a per value basis

Reply:
or are we thinking of having an event platform which has its own discovery independent of the platform discovery? So in addition to the switch entity we discover for a given value, we also discover an `event` instance?

Reply:
Yeah, we can create an event instance triggered by discovery.

Why can't we scope this per Value?

Reply:
I misunderstood in the first comment. Should we create a separate discovery scheme for events or can we just create the instance in the node ready callback when the assumed_state property of the discovery info is true?

Reply:
let me know what you think of the changes I made. We are now capturing discovery info anywhere where assumed_state is true and using that to watch value updates